include/nbl/builtin/hlsl/numeric_limits.hlsl

```diff
@@ -0,0 +1,202 @@
+    // 	identifies the denormalization style used by the floating-point type
+    static const float_denorm_style has_denorm = is_integer ? float_denorm_style::denorm_absent : float_denorm_style::denorm_present;
+    // identifies the floating-point types that detect loss of precision as denormalization loss rather than inexact result
+    static const bool has_denorm_loss = !is_integer;
+    // identifies the rounding style used by the type
+    static const float_round_style round_style = is_integer ? float_round_style::round_toward_zero : float_round_style::round_to_nearest;
+    // identifies the IEC 559/IEEE 754 floating-point types
+    static const bool is_iec559 = !is_integer;
+    // identifies types that represent a finite set of values
+    static const bool is_bounded = true;
+
+    // TODO verify this
+    // identifies types that handle overflows with modulo arithmetic
+    static const bool is_modulo = is_integer && !is_signed;
+
+    // number of radix digits that can be represented without change
+    static const int digits = !is_integer ? float_digits : (size - is_signed);
+    // number of decimal digits that can be represented without change
+    static const int digits10 = !is_integer ? float_decimal_digits : (size * 2 + size / 4 + int(!is_signed) * (size / 4););
+
+    // number of decimal digits necessary to differentiate all values of this type
+    static const int max_digits10 = !is_integer ? float_decimal_max_digits : 0;
+
+    // the radix or integer base used by the representation of the given type
+    static const int radix = 2;
+
+    // one more than the smallest negative power of the radix that is a valid normalized floating-point value
+    static const int min_exponent = !is_integer ? float_min_exponent : 0;
+    // the smallest negative power of ten that is a valid normalized floating-point value
+    static const int min_exponent10 = !is_integer ? float_min_decimal_exponent : 0;
+    // 	one more than the largest integer power of the radix that is a valid finite floating-point value
+    static const int max_exponent = !is_integer ? float_max_exponent : 0;
+    // the largest integer power of 10 that is a valid finite floating-point value
+    static const int max_exponent10 = !is_integer ? float_max_decimal_exponent : 0;
+
+    // identifies types which can cause arithmetic operations to trap
+    static const bool traps = false;
+    // identifies floating-point types that detect tinyness before rounding
+    static const bool tinyness_before = false;
+
+    static T min() { return num_definitions<T>::MIN; }
+    static T max() { return num_definitions<T>::MAX; }
+    static T lowest() { return -max(); }
+    static T denorm_min() { return num_definitions<T>::DENORM_MIN }
+    
+    static T epsilon() { return num_epsilon; }
+    static T round_error() { return T(0.5); }
+    static T infinity() { return T(0); }
+    static T quiet_NaN() { return T(0); }
+    static T signaling_NaN() { return T(0); }
```

Comment:
because of DXC not having constexpr functions, we need to break with STL and declate the functions as variables